sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java

```diff
@@ -0,0 +1,101 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.playtika.sleuth.camel;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.management.event.ExchangeCreatedEvent;
+import org.apache.camel.support.EventNotifierSupport;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanTextMap;
+import org.springframework.cloud.sleuth.Tracer;
+import org.springframework.cloud.sleuth.instrument.messaging.MessagingSpanTextMapExtractor;
+import org.springframework.cloud.sleuth.instrument.messaging.MessagingSpanTextMapInjector;
+import org.springframework.cloud.sleuth.util.SpanNameUtil;
+
+import java.util.EventObject;
+
+import static com.playtika.sleuth.camel.SleuthCamelConstants.FROM_CAMEL;
+
+@Slf4j
+@AllArgsConstructor
+public class CreatedEventNotifier extends EventNotifierSupport {
+
+    private static final String MESSAGE_COMPONENT = "camel";
+
+    private final Tracer tracer;
+    private final MessagingSpanTextMapExtractor spanExtractor;
+    private final MessagingSpanTextMapInjector spanInjector;
+
+    @Override
+    public void notify(EventObject event) {
+        log.trace("Caught ExchangeCreatedEvent, processing...");
+        ExchangeCreatedEvent exchangeCreatedEvent = (ExchangeCreatedEvent) event;
+        Exchange exchange = exchangeCreatedEvent.getExchange();
+        Endpoint endpoint = exchange.getFromEndpoint();
+        CamelMessageTextMap carrier = new CamelMessageTextMap(exchange.getIn());
+
+        Span span = getSpan(carrier, endpoint);
+        span.tag(FROM_CAMEL, "true");
+
+        spanInjector.inject(span, carrier);
+    }
+
+    private Span getSpan(CamelMessageTextMap carrier, Endpoint endpoint) {
+        Span spanFromMessage = buildSpan(carrier);
+        if (spanFromMessage != null) {
+            log.debug("Built span from message - {}. Assuming it is remote one. Continuing it...", spanFromMessage);
+            tracer.continueSpan(spanFromMessage);
+            spanFromMessage.logEvent(Span.SERVER_RECV);
+            return spanFromMessage;
+        } else {
```

Comment:
you can remove else

Reply:
Actually it was w\o else and I consider that code as less readable as we have now.